docs/tasks/tls/managing-tls-in-a-cluster.md

```diff
@@ -0,0 +1,189 @@
+---
+title: Managing TLS Certificates in a Cluster
+assignees:
+- mikedanese
+- beacham
+- liggit
+---
+
+* TOC
+{:toc}
+
+## Overview
+
+Every Kubernetes has a cluster root Certificate Authority (CA). The CA is
+generally used by cluster components to validate the API server's certificate,
+by the API server to validate kubelet client certificates, etc. To support
+this, the CA certificate bundle is distributed to every node in the cluster and
+is distributed as a secret attached to default service accounts.  Optionally,
+your workloads can use this CA to establish trust. Your application can request
+a certificate signing using the `certificates.k8s.io` API using a protocol
+that is similar to the [ACME draft](https://letsencrypt.github.io/acme-spec/).
+
+## Trusting TLS in a Cluster
+
+Trusting the cluster root CA from an application running as a pod usually
+requires some extra application configuration. You will need to add the CA
+certificate bundle to the list of CA certificates that the TLS client or server
+trusts. For example, you would do this with a golang TLS config by parsing the
+certificate chain and adding the parsed certificates to the `Certificates` field
+in the [`tls.Config`](https://godoc.org/crypto/tls#Config) struct.
+
+The CA certificate bundle is automatically mounted into pods using the default
+service account at the path `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`.
+If you are not using the default service account, ask a cluster administrator to
+build a configmap containing the certificate bundle that you have access to use.
+
+## Requesting a Certificate
+
+The following section demonstrates how to create a TLS certificate for a
+Kubernetes service accessed through DNS.
+
+### Step 0. Download and install CFSSL
+
+The cfssl tools used in this example can be downloaded at
+[https://pkg.cfssl.org/](https://pkg.cfssl.org/).
+
+### Step 1. Create a Certificate Signing Request
+
+Generate a private key and certificate signing request (or CSR) by running the
+following command:
+
+```console
+$ cat <<EOF | cfssl genkey - | cfssljson -bare server
+{
+  "hosts": [
+    "my-svc.my-namespace.svc.cluster.local",
+    "172.168.0.24",
+    "10.0.34.2"
+  ],
+  "CN": "my-svc.my-namespace.svc.cluster.local",
+  "key": {
+    "algo": "ecdsa",
+    "size": 256
+  }
+}
+EOF
+```
+
+Where `172.168.0.24` is the service's cluster IP and `10.0.34.2` is the pod's
```

Comment:
Not actionable for this PR since the ship has sailed here, but using IPs as identities like this and having them crop up in SANs doesn't feel right.